src/extras/DataUtils.js

```diff
@@ -8,144 +8,154 @@ class DataUtils {
 
 	static toHalfFloat( val ) {
 
+		if ( ! this._floatView ) this._generateTables();
```

Comment:
I'm going to convert this to plain functions. See https://github.com/mrdoob/three.js/issues/24199#issuecomment-1151252323

Reply:
On the polluting namespace thing, in this case I think we do need some kind of flag for `_generateTables()` because of all for loops, it needs to be generated lazily.

In the singleton approach above it's using the existing `_floatView` (kind of duck typing, common with singleton patterns, I know 😬). I could alternatively check the contents of `_floatView`? For example:

```js
function toHalfFloat( val ) {

	if ( _floatView.length === 0 ) _generateTables();
...
```

Or a `needsUpdate` kind of thing?

```js
function toHalfFloat( val ) {

	if ( needsUpdate === true ) _generateTables();
...
```

Are there other examples of classes in three.js that are lazy loaded?

Reply:
I'm also going to test @marcofugaro's pure annotation suggestion https://github.com/mrdoob/three.js/issues/24199#issuecomment-1151249404, maybe it'll work with tree-shaking if the for loops are in a `_generateTables()` function...

Reply:
@pschroen I don't think there's the need for the `if ( ... )`, the generation could be done outside of the functions like it is now. It's pretty fast anyway.

Reply:
@marcofugaro Right, so I've almost got it working, I can't use `/*@__PURE__*/ _generateTables();` on it's own because then there's no reference to it and it gets removed, and if I do the following it will be regenerated every-time.

```js
function toHalfFloat( val ) {

	const tables = _generateTables();
...
```

Ideally we'd want a cached version of the lookup table so it's only generated once.

Reply:
> the generation could be done outside of the functions like it is now. It's pretty fast anyway.

So are you're saying you're cool with generating the lookup table on every call to `toHalfFloat` and `fromHalfFloat`? If so, I got that working...

Reply:
> I can't use /*@__PURE__*/ _generateTables();

Can you do something like `const { table1, table2, ... } = /*@__PURE__*/ _generateTables();`?

Reply:
Ah good idea, I'll try that!